packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenView.kt

```diff
@@ -0,0 +1,41 @@
+package expo.modules.splashscreen
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.RelativeLayout
+
+@SuppressLint("ViewConstructor")
+class SplashScreenView(
+  context: Context,
+  resizeMode: SplashScreenImageResizeMode,
+  splashScreenConfigurator: SplashScreenConfigurator
+) : RelativeLayout(context) {
+  private val imageView: ImageView
+
+  init {
+    imageView = ImageView(context).also { view ->
+      view.layoutParams = LayoutParams(
+        LayoutParams.MATCH_PARENT,
+        LayoutParams.MATCH_PARENT
+      ).also { layoutParams ->
+        layoutParams.addRule(CENTER_IN_PARENT, TRUE)
+      }
+    }
+
+    layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+    setBackgroundColor(splashScreenConfigurator.getBackgroundColor(context))
```

Comment:
Should/shouldn't this (`setBackgroundColor` and `adjustViewBounds`) be set in `splashScreenConfigurator`? Why expose `getBackgroundColor` from `splashScreenConfigurator` if all the configuration could be done in one opaque place?

Reply:
I've adjusted a little bit classes here. There's no longer any `SplashScreenConfigurator`, but instead I have `SplashScreenResourcesProvider` that plays nicer with current module design 😉